src/js/letters/containers/AddressSection.jsx

```diff
@@ -142,10 +142,12 @@ export class AddressSection extends React.Component {
       // Reset all the error messages in case they go to edit again; should be pointless
       errorMessages: {}
     });
+    window.dataLayer.push({ event: 'letter-update-address-submit' });
     this.props.saveAddress(this.state.editableAddress);
   }
 
   handleCancel = () => {
+    window.dataLayer.push({ event: 'letter-update-address-cancel' });
```

Comment:
So I guess the intent with this and `'letter-update-address-started'` is to log each time a user hits edit or cancel, understanding that there may be several of these per user? If so, 👍